### ISSUE TITLE
Use uiop:define-package in package section

### ISSUE THREAD

**Issue by kilianmh** [LispCookbook/cl-cookbook#539]
`uiop define-package` is more flexible with additional clauses and more convenient during development (e.g. when symbols change).

I want to add some examples in the packages page for `uiop:define-package`.

Currently there are only examples with `cl:defpackage`. 
My question is: Should we add a few examples with `uiop:define-package` but keep it mostly with `cl:defpackage` or should we convert most package definitions to `uiop:define-package` and only showcase few examples of `cl:defpackage`?

**Comment by vindarel** [comment]
Hi, I think it's great if we mention it (even required), maybe not in the introductory examples. defpackage works when we begin, after all.

Shall we add a second "packages (advanced)" page?

Presenting define-package's other clauses, the difference in behavior, also shadowing imports would be very useful.

**Comment by Jach** [comment]
+1 on keeping things the normal `cl:defpackage` unless otherwise needed.

Extending the existing page would probably be best, as it's already bringing up more advanced topics by the end with locked packages. A simple example I like is in the generic-cl docs: https://gutev.dev/generic-cl/#using_specific_interface

Somewhat related, do you think a section on ASDF package-inferred-system as in #264 would best live in the Systems page, the Packages page, or a separate page? I was thinking it might be useful to have a section in the Packages page on just the style of having a main package(s).lisp file vs. one file, one defpackage, vs. going further with the package-inferred-system feature. Perhaps it's something more relevant for a separate issue as part of some new "Project Organization" page that can collect the various styles and tradeoffs that projects do in the wild.

**Comment by vindarel** [comment]
yeah I can see that in a new page, otherwise it will trip up newcomers too easily. A "Project Organization" page would fulfill a demand I think.
